### DISCUSSION TITLE
Tengo un Segmentation Fault debido a algun problema de memoria

### DISCUSSION THREAD

**Discussion by andresreopo** [IIC2133-PUC/2024-1#116] [Tarea 1] [answered]
Hola, hice el árbol de la parte 2.3 (YEAR-DISTRICT) usando concatenación de las llaves con un float. Resulta que haciendo eso y ejecutando los testcases cortos (ayudantes y futurama) me funciona bien, pero a la hora de usarlo con el testase largo (wikichilean) me tira un Segmentation Fault y no sé como arreglarlo. Aquí va mi output con valgrind.

![image](https://github.com/IIC2133-PUC/2024-1/assets/127421678/3c9cb46b-6c46-4784-b376-1b708f7d1882)

En el output no tengo nada porque el error está al insertar los nodos year_dis, como dice el valgrind.

Quizas tengo un stackoverflow como dice valgrind? No lo sé, por eso abro esta issue.

Muchas gracias de antemano

**Comment by PabloNavarroY** [comment, marked as answer]
En la linea que te aparece en el error de valgrind estás leyendo el atributo de un puntero, pero no revisaste si el puntero es distinto de NULL.

**Reply by PabloNavarroY** [reply]
Para dilucidar el proceso por el cuál llegué a esa conclusión:
En la imagen dice
```
Address 0x8 is not stack'd, malloc'd or (recently) free'd
```
y 0x8 es el valor de NULL.

**Reply by andresreopo** [reply]
Lo logré debuggear!!
Efectivamente estaba accediendo a un atributo de un nodo null jajaj

Muchas gracias!!!
